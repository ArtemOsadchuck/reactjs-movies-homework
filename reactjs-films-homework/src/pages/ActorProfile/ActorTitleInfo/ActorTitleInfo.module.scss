$background-color: rgb(37, 37, 37);
$h2-loading-color: rgb(121, 121, 121);
$wite-color: rgb(255, 255, 255);

.titleWrapper {
  color: $wite-color;
  display: flex;
  padding-right: 1.5rem;
  width: 100%;

  @media screen and (max-width: 1200px) {
    align-items: center;
    flex-wrap: wrap;
    justify-content: center;
  }
}

.titleImageWrapper {
  display: flex;
  float: left;
  height: 29rem;
  padding-left: 1.5rem;
  width: 22rem;

  .titleImage {
    border-radius: 5px;
    flex-shrink: 0;
  }
}

.titleInfoWrapper {
  align-items: flex-start;
  display: flex;
  flex-direction: column;
  justify-content: space-around;
  width: 68%;

  .Info {
    margin-top: 1rem;
    text-align: start;

    strong {
      display: block;
      margin-bottom: 0.2rem;
      text-align: start;
    }

    p {
      text-align: start;
    }

    .biographyTitle {
      display: block;
      margin-bottom: 1rem;
    }

    .biography {
      width: 80vw;
    }
  }

  @media screen and (max-width: 1200px) {
    padding: 0 1.5rem;
    width: 100%;
  }
}
