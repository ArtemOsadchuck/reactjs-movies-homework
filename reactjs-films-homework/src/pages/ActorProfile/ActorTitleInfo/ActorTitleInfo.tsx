import React from 'react';
import styles from './ActorTitleInfo.module.scss';

export interface IActorTitle {
  props: IActorTitleProps;
}
export interface IActorTitleProps {
  also_known_as?: string[];
  biography: string;
  birthday: string;
  gender: number;
  id: number;
  known_for_department?: string;
  name: string;
  place_of_birth: string;
  profile_path: string;
}

const ActorTitle: React.FC<IActorTitle> = ({ props }) => {
  const { birthday, name, profile_path, biography, place_of_birth } = props;
  const urlImg = `https://image.tmdb.org/t/p/w500/${profile_path}`;
  return (
    <div className={styles.titleWrapper}>
      <div className={styles.titleImageWrapper}>
        {props.profile_path ? (
          <img className={styles.titleImage} src={urlImg} alt={name} />
        ) : null}
      </div>
      {name ? (
        <article className={styles.titleInfoWrapper}>
          <h1 className={styles.titleName}>{name}</h1>
          <div className={styles.Info}>
            <strong>Birthday</strong>
            <p>{birthday}</p>
          </div>
          <div className={styles.Info}>
            <strong>place of birth</strong>
            <p>{place_of_birth}</p>
          </div>
          <div className={styles.Info}>
            <strong className={styles.biographyTitle}>Biography:</strong>
            <p className={styles.biography}>{biography}</p>
          </div>
        </article>
      ) : null}
    </div>
  );
};
export default ActorTitle;
