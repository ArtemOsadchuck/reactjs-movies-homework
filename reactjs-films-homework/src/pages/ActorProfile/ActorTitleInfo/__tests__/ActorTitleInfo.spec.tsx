import React from 'react';
import { render, screen } from '@testing-library/react';

import ActorTitleInfo, { IActorTitleProps } from '../ActorTitleInfo';
import ActorTitleInfoMocks from '../mocks';

import StoreWrapper from '../../../../__testsUtils__/storeHoc';

const getComponent = (props: IActorTitleProps) => (
  <StoreWrapper>
    <ActorTitleInfo actorInfo={props} />
  </StoreWrapper>
);

describe('ActorTitleInfo', () => {
  it('should renders correctly', () => {
    const { asFragment } = render(getComponent(ActorTitleInfoMocks));
    const fragment = asFragment();

    const date = screen.getByText(/1937-06-01/i);

    expect(fragment).toMatchSnapshot();
    expect(date).toBeInTheDocument();
  });
});
