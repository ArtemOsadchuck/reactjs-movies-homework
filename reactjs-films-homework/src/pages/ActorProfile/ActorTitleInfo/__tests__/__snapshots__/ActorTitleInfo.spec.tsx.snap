// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ActorTitleInfo ActorTitleInfo snapshot 1`] = `[Function]`;
